network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractLoadingLimits.java

```diff
@@ -73,8 +73,13 @@ public T setPermanentLimit(double permanentLimit) {
 
     @Override
     public Collection<LoadingLimits.TemporaryLimit> getTemporaryLimits() {
-        return attributes.getTemporaryLimits() == null ? Collections.emptyList()
-                                                       : attributes.getTemporaryLimits().values().stream().map(TemporaryLimitImpl::new).collect(Collectors.toUnmodifiableList());
+        if (attributes.getTemporaryLimits() == null) {
+            return Collections.emptyList();
+        }
+        // make sure limits are provided by descending durations
+        TreeMap<Integer, TemporaryLimitAttributes> sortedLimitAttributes = new TreeMap<>(ACCEPTABLE_DURATION_COMPARATOR);
```

Comment:
There is already a TreeMap in the TemporaryLimitAttributes which is configured for natural order sorting. You just have to change TemporaryLimitAttributes for implementing Comparable<TemporaryLimitAttributes> and it should be ok without any copy.

Reply:
yes done